maths/two-sum.rb

```diff
@@ -0,0 +1,12 @@
+# @param {Integer[]} nums
+# @param {Integer} target
+# @return {Integer[]}
+def two_sum(nums, target)
+    nums_hash = Hash.new
+    
+    nums.each_with_index do |val, ind|
+      complement = target - val
+      return [ind, nums_hash[complement]] if nums_hash[complement]
+      nums_hash[val] = ind
+    end
+end
```

Comment:
We have this algorithm:
- https://github.com/TheAlgorithms/Ruby/blob/4fa7e1a672737bf9ca06381f003e2444d2b8cdb6/data_structures/arrays/two_sum.rb
- https://github.com/TheAlgorithms/Ruby/blob/4fa7e1a672737bf9ca06381f003e2444d2b8cdb6/data_structures/arrays/two_sum_ii.rb
- https://github.com/TheAlgorithms/Ruby/blob/4fa7e1a672737bf9ca06381f003e2444d2b8cdb6/data_structures/hash_table/two_sum.rb

Reply:
Good catch, thank you. I'll start now.

Reply:
Thanks. I'll close this PR for now.